tests/glibc/getopt.c

```diff
@@ -12,25 +17,27 @@ int main() {
 		{NULL, no_argument, NULL, 0}
 	};
 
-	int test_argc = 7;
+	int test_argc = 6;
 
 	char *test_argv[] = {
 		"dummy",
 		"--foo",
 		"abc",
-		"-b",
-		"abc",
+		"-babc",
 		"abc",
 		"abc"
 	};
 
+
 	int c;
 	c = getopt_long(test_argc, test_argv, shortopts, longopts, NULL);
+	debug(c);
 	assert(c == 'f');
 
 	c = getopt_long(test_argc, test_argv, shortopts, longopts, NULL);
-	assert(!strcmp(optarg, "abc"));
+	debug(c);
 	assert(c == 'b');
+	assert(!strcmp(optarg, "abc"));
```

Comment:
cc @no92, these changes were necessary to get the `getopt` test to run on `glibc` but it implies that our implementation in `mlibc` is broken.

Reply:
Weird, they work for me on both mlibc and glibc. Also, I remember checking against the man pages, and our mlibc behavior should match that, IIRC.

As far as I can tell, these changes should not really change the behavior anyways other than using concatenated short options.

Also, #438 touches this test and implements missing code paths in `getopt_long`.

Reply:
With the double colon, the man page doesn't say that it should eat the next argument into optarg. What glibc version do you use?